DOL.WHD.Section14c.Web/src/modules/components/sectionAppInfo/sectionAppInfoTemplate.html

```diff
@@ -97,14 +97,15 @@
 
 
   <fieldset class="dol-form-sub-section">
-    <legend>
-      <h3><span id="applicationContactLabel">Application Contact Person</span>
-      <helplink aria-controls="applicationContactPersonHelp" aria-describedby="applicationContactLabel"></helplink>
-      </h3>
+    <legend class="hide">
+      <span id="applicationContactLabel">Application Contact Person</span>
     </legend>
+    <div class="dol-legend-h3-help">
+      <h3 aria-hidden="true">Application Contact Person</span></h3>
+      <helplink aria-controls="applicationContactPersonHelp" aria-describedby="applicationContactLabel"></helplink>        
+    </div>
```

Comment:
This is interesting.  Making the `<legend>` visible to screen readers and hiding the `<h3>`, and reversing it for the visual rendering, right?  Is this to do with making the `legend` more contextual?  This is neat stuff.

Reply:
Actually we had to do this in the case of a legend with help button because the help button needs to be inline with the text. We can't have the help button in the legend, because the screen reader duplicates reading it in that case. We can't have it after a visible legend, because the legend is a block element and the button won't flow with the text (Shannon and I spent quite some time trying to magic a way around it, but no luck). So we settled on having a legend for the screen reader only, and visible text hidden from screen readers (with button flowing with the text) for sighted users. In the case of a legend that has a h3, I made the h3 an inline element so the button can flow with it. Regular legends that DON'T have a help button DON'T get all this complicated hidden/shown stuff. It's all to make a help button flow with the text. Whew!